Libraries/WebSocket/RCTSRWebSocket.m

```diff
@@ -492,7 +492,22 @@ - (void)didConnect
     CFHTTPMessageSetHeaderFieldValue(request, CFSTR("Sec-WebSocket-Protocol"), (__bridge CFStringRef)[_requestedProtocols componentsJoinedByString:@", "]);
   }
 
-  CFHTTPMessageSetHeaderFieldValue(request, CFSTR("Origin"), (__bridge CFStringRef)(_requestedOptions[@"origin"] ?: _url.RCTSR_origin));
+  if (_requestedHeaders) {
+    
+    if (![_requestedHeaders objectForKey:@"origin"]) {
+      CFHTTPMessageSetHeaderFieldValue(request, CFSTR("origin"), (__bridge CFStringRef)(_url.RCTSR_origin));
+    }
+    
+    [_requestedHeaders enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+       if ([obj isKindOfClass:[NSString class]]) {
```

Comment:
Could just use `[obj description]` to convert it to a string here. That will do the right thing for numbers, which are the most likey kind of wrong value to get passed in.